src/extras/inputs/joystickdouble-input.js

```diff
@@ -0,0 +1,165 @@
+import { Delta, Input } from './input.js';
+import { Joystick } from './joystick.js';
+
+class JoystickDoubleInput extends Input {
+    /**
+     * @type {Map<number, { x: number, y: number, left: boolean }>}
+     * @private
+     */
+    _pointerData = new Map();
+
+    /**
+     * @type {Joystick}
+     * @private
+     */
+    _leftJoystick;
+
+    /**
+     * @type {Joystick}
+     * @private
+     */
+    _rightJoystick;
+
+    /**
+     * @override
+     */
+    deltas = {
+        leftStick: new Delta(2),
+        rightStick: new Delta(2)
+    };
+
+    constructor() {
+        super();
+
+        this._leftJoystick = new Joystick();
+        this._rightJoystick = new Joystick();
+
+        this._onPointerDown = this._onPointerDown.bind(this);
+        this._onPointerMove = this._onPointerMove.bind(this);
+        this._onPointerUp = this._onPointerUp.bind(this);
+    }
+
+    get leftJoystick() {
+        return this._leftJoystick;
+    }
+
+    get rightJoystick() {
+        return this._rightJoystick;
+    }
+
+    /**
+     * @private
+     * @param {PointerEvent} event - The pointer event.
```

Comment:
Nitpick: we generally use a convention of putting visibility tags after other tags. Again, adhere to what you see elsewhere.